mongo/controller.go

```diff
@@ -0,0 +1,41 @@
+package mongo
+
+import (
+	"fmt"
+	"log"
+	"context"
+	"go.mongodb.org/mongo-driver/mongo"
+    "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Controller struct {
+	Client *mongo.Client
+}
+
+func NewController(url string) Controller {
+
+	var newController Controller;
```

Comment:
This variable isn't needed, and if it was, it should be initialized with `controller := Controller{}`.
You can remove this variable and just write `return Controller{Client: client}` at the end.